pkg/controllers/provisioning/scheduling/scheduler.go

```diff
@@ -240,11 +241,16 @@ func (s *Scheduler) Solve(ctx context.Context, pods []*corev1.Pod) Results {
 			delete(errors, k)
 		}
 	}
-	return Results{
+	var results = Results{
 		NewNodeClaims: s.newNodeClaims,
 		ExistingNodes: s.existingNodes,
 		PodErrors:     errors,
 	}
+	results.TruncateInstanceTypes(MaxInstanceTypes)
```

Comment:
nit: prefer walrus assignment
```suggestion
    results := Results{
		NewNodeClaims: s.newNodeClaims,
		ExistingNodes: s.existingNodes,
		PodErrors:     errors,
	}
	results.TruncateInstanceTypes(MaxInstanceTypes)
```